__has_include(<new>) not working on ESP8266 Platformio builds using old framework

Hey there,

When I attempt to compile the [DallasTemperature.ino ](https://github.com/thorrak/OneWireNg/blob/master/examples/DallasTemperature/DallasTemperature.ino) example for an ESP8266 using `platform = espressif8266@2.6.3` (the latest version of the 2.x ESP8266 platform available on Platformio), I get the following error:

```
In file included from .pio/libdeps/esp12e/OneWireNg/src/OneWireNg.h:19:0,
                 from .pio/libdeps/esp12e/OneWireNg/src/OneWireNg_BitBang.h:16,
                 from .pio/libdeps/esp12e/OneWireNg/src/OneWireNg_BitBang.cpp:15:
.pio/libdeps/esp12e/OneWireNg/src/platform/Platform_New.h: In function 'void* operator new(size_t, void*)':
.pio/libdeps/esp12e/OneWireNg/src/platform/Platform_New.h:52:14: error: redefinition of 'void* operator new(size_t, void*)'
 inline void *operator new(size_t sz, void *ptr) NOEXCEPT {
              ^
In file included from /Users/jbeeler/.platformio/packages/toolchain-xtensa@2.40802.200502/xtensa-lx106-elf/include/c++/4.8.2/bits/stl_construct.h:59:0,
                 from /Users/jbeeler/.platformio/packages/toolchain-xtensa@2.40802.200502/xtensa-lx106-elf/include/c++/4.8.2/bits/stl_tempbuf.h:60,
                 from /Users/jbeeler/.platformio/packages/toolchain-xtensa@2.40802.200502/xtensa-lx106-elf/include/c++/4.8.2/bits/stl_algo.h:62,
                 from /Users/jbeeler/.platformio/packages/toolchain-xtensa@2.40802.200502/xtensa-lx106-elf/include/c++/4.8.2/algorithm:62,
                 from /Users/jbeeler/.platformio/packages/framework-arduinoespressif8266@3.20704.0/cores/esp8266/Arduino.h:238,
                 from .pio/libdeps/esp12e/OneWireNg/src/platform/Platform_Delay.h:17,
                 from .pio/libdeps/esp12e/OneWireNg/src/OneWireNg_BitBang.cpp:13:
/Users/jbeeler/.platformio/packages/toolchain-xtensa@2.40802.200502/xtensa-lx106-elf/include/c++/4.8.2/new:109:14: error: 'void* operator new(std::size_t, void*)' previously defined here
 inline void* operator new(std::size_t, void* __p) _GLIBCXX_USE_NOEXCEPT
              ^
```


Digging into it, it appears that the `__has_include` [check in Platform_New.h](https://github.com/pstolarz/OneWireNg/blob/master/src/platform/Platform_New.h#L30-L33
) is failing. 

Best that I can tell `__has_include` is #define'd as `0` in this version of the xtensa toolchain, meaning that simply testing if it is defined is not proof-positive that it actually available. 

I've produced a MCVE here: https://github.com/thorrak/onewireng_platform_new_issue -- `[env:esp8266_2.6.3]` is broken, while `[env:esp8266_current]` works.

--- a/src/platform/Platform_New.h
+++ b/src/platform/Platform_New.h
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2021 Piotr Stolarz
+ * Copyright (c) 2021,2022 Piotr Stolarz
  * OneWireNg: Arduino 1-wire service library
  *
  * Distributed under the 2-clause BSD License (the License)
@@ -27,11 +27,15 @@
 # define NOEXCEPT throw()
 #endif
 
-#if !defined(CONFIG_CPP_NEW_ALT) && \
-    (defined(__has_include) && __has_include(<new>))
-# include <new>
+
+#ifndef CONFIG_CPP_NEW_ALT
+# if defined(__has_include) && __has_include(<new>)
+#  include <new>
+# elif !defined(_NEW)
+#  warning "<new> header not detected; try to use CONFIG_CPP_NEW_ALT in case of problems"
+# endif
 #else
-# if defined(CONFIG_CPP_NEW_ALT) && defined(_NEW)
+# ifdef _NEW
 #  warning "CONFIG_CPP_NEW_ALT ignored to avoid conflict with already included <new> header"
 # else
 #  include <stdlib.h>

--- a/src/platform/Platform_New.h
+++ b/src/platform/Platform_New.h
@@ -28,22 +28,20 @@
 #endif
 
 
-#ifndef CONFIG_CPP_NEW_ALT
-# if defined(__has_include) && __has_include(<new>)
-#  include <new>
-# elif !defined(_NEW)
-#  warning "<new> header not detected; try to use CONFIG_CPP_NEW_ALT in case of problems"
+#if !defined(CONFIG_CPP_NEW_ALT) && \
+    (defined(__has_include) && __has_include(<new>))
+# include <new>
+#elif defined(_NEW)
+# ifdef CONFIG_CPP_NEW_ALT
+#  warning "CONFIG_CPP_NEW_ALT ignored to avoid conflict with already included <new> header"
 # endif
 #else
-# ifdef _NEW
-#  warning "CONFIG_CPP_NEW_ALT ignored to avoid conflict with already included <new> header"
-# else
-#  include <stdlib.h>
+# include <stdlib.h>
 
-#  if __cpp_aligned_new
+# if __cpp_aligned_new
 static_assert(alignof(max_align_t) >= __STDCPP_DEFAULT_NEW_ALIGNMENT__,
     "Alt. allocation implementation can't guarantee proper C++17 alignment");
-#  endif
+# endif
 
 inline void *operator new(size_t sz) {
     return malloc(sz);
@@ -66,7 +64,7 @@ inline void operator delete[](void *ptr) NOEXCEPT {
     free(ptr);
 }
 
-#  if __cpp_sized_deallocation
+# if __cpp_sized_deallocation
 inline void operator delete(void* ptr, size_t sz) NOEXCEPT {
     (void)sz;
     free(ptr);
@@ -76,7 +74,6 @@ inline void operator delete[](void* ptr, size_t sz) NOEXCEPT {
     (void)sz;
     free(ptr);
 }
-#  endif
 # endif
 #endif
 

